services/watson_service.py

```diff
@@ -0,0 +1,39 @@
+import os
+from services.clean_text import clean_text
+from ibm_watson import NaturalLanguageUnderstandingV1
+from ibm_cloud_sdk_core.authenticators import IAMAuthenticator
+from ibm_watson.natural_language_understanding_v1 import Features, SentimentOptions
+
+class WatsonService:
+    def __init__(self, text):
+        self.text = clean_text(text)
+
+
+    def convert_to_spotify_valence(self, sentiment_value):
+        if sentiment_value > 0.333:
+            return 1
+        elif sentiment_value < -0.333:
+            return 0
+        else:
+            return 0.5
+
+
+    def get_sentiment_value(self):
+        if len(self.text) > 20:
```

Comment:
I understand what this code is doing, but not exactly why. Does Watson need at least 20 characters in order to perform the sentiment analysis?

Reply:
Yes, Watson external API returns a 422 error if not enough characters for sentiment analysis.

Reply:
👍